### PR TITLE
Improve performance of the Instant Search queries

### PR DESCRIPTION
**Pull request by vinaysshenoy**

### CODE REVIEW

File: app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt [hunk 1]
```diff
@@ -3938,11 +3938,12 @@ class PatientRepositoryAndroidTest {
...
-        patient2WithCurrentFacilityAsAssignedFacility,
-        patientWithCurrentFacilityAsAssignedFacility,
-        patientWithAnotherFacilityAsAssignedFacility
-    )
+    assertThat(searchResults)
+        .containsExactly(
+            patientWithCurrentFacilityAsAssignedFacility,
+            patient2WithCurrentFacilityAsAssignedFacility,
+            patientWithAnotherFacilityAsAssignedFacility
+        ).inOrder()
```
Comment (msasikanth) [inline app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt:3946]: Correct me if I am wrong, will `contains` doesn't verify the order but passes even if there are more items than expected, `containsExactly` verifies that it has the same items as the expected but doesn't verify the order right?

File: app/src/main/java/org/simple/clinic/patient/PatientSearchResult.kt [hunk 2]
```diff
@@ -158,17 +158,17 @@ data class PatientSearchResult(
...
-            WHERE P.deletedAt IS NULL AND P.assignedFacilityId = :facilityId AND namePosition > 0
-            UNION
-            SELECT searchResult.*, 1 priority, INSTR(lower(P.fullName), lower(:name)) namePosition FROM 
-            PatientSearchResult searchResult
-            LEFT JOIN Patient P ON P.uuid = searchResult.uuid
-            WHERE P.deletedAt IS NULL AND P.assignedFacilityId != :facilityId AND namePosition > 0
-        )
+        SELECT 
+            searchResult.*, 
+            (
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/patient/PatientSearchResult.kt:163]: OMG, the easiest approach. Can't believe I missed it when implementing this feature 😅

**Review by msasikanth — APPROVED** [review]
LGTM 🚀

File: app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt (hunk 1 above)
Comment (vinaysshenoy) [reply app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt:3946]: Yup, that is correct.

Comment (vinaysshenoy) [reply app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt:3946]: The clue is in the assertion name itself, `contains`.

**Pull request merged**